From: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Subject: fbdev: have CONFIG_FB_DEFERRED_IO depend on CONFIG_MMU
Date: Sat, 1 Feb 2025 16:56:02 +0000

Frame buffer deferred I/O is entirely reliant on the page faulting
mechanism (and thus, an MMU) to function.

Express this dependency in the Kconfig, as otherwise randconfig could
generate invalid configurations resulting in build errors.

Link: https://lkml.kernel.org/r/e38046bc-fa35-4f4d-b0e8-00ac4637c725@lucifer.local
Signed-off-by: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202502020030.MnEJ847Z-lkp@intel.com/
Cc: David Hildenbrand <david@redhat.com>
Cc: Helge Deller <deller@gmx.de>
Cc: Jaya Kumar <jayakumar.lkml@gmail.com>
Cc: Kajtar Zsolt <soci@c64.rulez.org>
Cc: Maíra Canal <mcanal@igalia.com>
Cc: Mattew Wilcox <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 drivers/video/fbdev/core/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/video/fbdev/core/Kconfig~fb_defio-do-not-use-deprecated-page-mapping-index-fields-fix
+++ a/drivers/video/fbdev/core/Kconfig
@@ -135,6 +135,7 @@ config FB_SYSMEM_FOPS
 config FB_DEFERRED_IO
 	bool
 	depends on FB_CORE
+	depends on MMU
 
 config FB_DMAMEM_HELPERS
 	bool
_
